_posts/2023-08-30-spiffe.md

```diff
@@ -0,0 +1,364 @@
+Deploy it using the following command:
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/nicholasM95/spiffe-demo/main/setup/clusterissuer.yaml
+```
+
+### Verify installation
+
+Verify if all pods have started and check whether the secret csi-driver-spiffe-ca has been created.
+
+<img src="{{ '/img/2023-08-30-spiffe/spiffe-1.png' | prepend: site.baseurl }}" alt="Verify spiffe installation" class="image fit" style="margin:0px auto; max-width:100%">
+
+## Configure AWS
+### Create the s3 bucket
+
+Create an S3 bucket with the name 'demospiffe' and upload a text file named 'hello.txt'. Our application will read this file later.
+
+<img src="{{ '/img/2023-08-30-spiffe/spiffe-2.png' | prepend: site.baseurl }}" alt="Create S3 bucket" class="image fit" style="margin:0px auto; max-width:100%">
+
+### Create a trusted anchor
+In the context of AWS, a trusted anchor might involve setting up a trust relationship with an external identity provider or certificate authority. This ensures that the SPIFFE identities generated within the Kubernetes cluster can be verified and trusted by AWS services when defining policies, access controls, or other security-related configurations.
+
+
+To achieve this, we need a CA certificate that is utilized by SPIFFE on our Kubernetes cluster. Use the following command to obtain the CA certificate:
+
+```bash
+kubectl get secret csi-driver-spiffe-ca -o jsonpath='{.data}'
+```
+
+Copy the value of ca.crt and decode it using base64.
+Example: 
+
+```bash
+echo 'copied-value' | base64 -d
+```
+
+Navigate to the IAM section within AWS and access the "Roles" tab.
+Scroll down and choose "Roles Anywhere."
+Ensure that you are still within the desired region.
+Generate a trust anchor.
+Opt for the external certificate bundle option.
+Paste the decoded value of `ca.crt` from the secret.
+
+<img src="{{ '/img/2023-08-30-spiffe/spiffe-3.png' | prepend: site.baseurl }}" alt="Create Trust anchor" class="image fit" style="margin:0px auto; max-width:100%">
+
+
+### Create a trusted profile
+
+A trusted profile establishes trust, defines access controls, and enables secure interaction between SPIFFE-issued identities within a Kubernetes cluster and AWS resources. It ensures a smooth and secure integration between these two environments while maintaining a solid security posture.
+
+We require a policy that allows our application, operating in the 'spiffe' namespace with service account 'file-read-app', to gain read access to files within our designated bucket, all within the 'demo.home.cluster' cluster.
+
+```json
+{
+    "Version": "2012-10-17",
+	  "Statement": [
+		    {
+			      "Action": "s3:GetObject",
+			      "Condition": {
+				        "StringLike": {
+					          "aws:PrincipalTag/x509SAN/URI": "spiffe://demo.home.cluster/ns/spiffe/sa/file-read-app"
+				        }
+			      },
+			      "Effect": "Allow",
+			      "Resource": "arn:aws:s3:::demospiffe/*",
```

Comment:
If you decide to go with my suggestion (your choice 😄), maybe change the 'demospiffe' part to something like <BUCKET_NAME>